### ISSUE TITLE
ManualParameter calls validate twice

### ISSUE THREAD

**Issue by peendebak** [QCoDeS/Qcodes#754]
The qcodes `ManualParameter` calls validate twice for a single set.
``` python
import qcodes
from qcodes.utils.validators import Validator

class MyValidator(Validator):

    def __init__(self, min_value=-float("inf"), max_value=float("inf")):
        pass

    def validate(self, value, context=''):
        print('validate!')

    is_numeric=True

p = qcodes.ManualParameter('p', initial_value=0, vals=MyValidator())
p.set(10)
```
The output is:
```
validate!
validate!
```
The validator should only be called once.

@WilliamHPNielsen @jenshnielsen

**Comment by WilliamHPNielsen** [comment]
I guess this is fixed by #651, which should be merged soon.

**Comment by WilliamHPNielsen** [comment]
Hmm... it is not. I'll look in to this.
